pkg/cli/mode/mode.go

```diff
@@ -4,18 +4,22 @@ package mode
 
 import "src.elv.sh/pkg/ui"
 
-// ModeLine returns a text styled as a modeline.
-func ModeLine(content string, space bool) ui.Text {
+// modeLine returns a text styled as a modeline.
+func modeLine(content string, space bool) ui.Text {
 	t := ui.T(content, ui.Bold, ui.FgWhite, ui.BgMagenta)
 	if space {
 		t = ui.Concat(t, ui.T(" "))
 	}
 	return t
 }
 
-// ModePrompt returns a callback suitable as the prompt in the codearea of a
+// modePrompt returns a callback suitable as the prompt in the codearea of a
 // mode widget.
-func ModePrompt(content string, space bool) func() ui.Text {
-	p := ModeLine(content, space)
+func modePrompt(content string, space bool) func() ui.Text {
+	p := modeLine(content, space)
 	return func() ui.Text { return p }
 }
+
+// Prompt is solely for the benefit of pkg/edit/minibuf.
+// TODO: Figure out how to make that a CLI widget so we don't have to publicly export modePrompt.
```

Comment:
The prompt can't be easily turned into a CLI widget with the current widget abstraction - even if it could, it seems overkill. So it is intentional that this is exported, it's not something that needs to be fixed.

What you're doing here - de-exporting the function and name the exported alias without the `Mode` suffix - is fine though. Personally I'm not too bothered by golint's warning, but I'm also not against fixing them. Just remove the TODO here, and move the docstring of `modePrompt` here so that it shows up nicely in godoc.

Reply:
Done.

I too am not particularly bothered by some of golint's warnings. But the stuttering warning this change addresses does suggest an API problem; hence the "TODO" that I've removed in response to your feedback.